be/src/storage/lake/tablet_manager.cpp

```diff
@@ -348,6 +421,87 @@ StatusOr<TabletMetadataPtr> TabletManager::get_tablet_metadata(const string& pat
     return ptr;
 }
 
+StatusOr<TabletMetadataPtr> TabletManager::get_single_tablet_metadata(int64_t tablet_id, int64_t version,
+                                                                      bool fill_cache) {
+    auto tablet_path = tablet_metadata_location(tablet_id, version);
+    if (auto ptr = _metacache->lookup_tablet_metadata(tablet_path); ptr != nullptr) {
+        return ptr;
+    }
+
+    auto path = aggregate_tablet_metadata_location(tablet_id, version);
+    RandomAccessFileOptions opts{.skip_fill_local_cache = !fill_cache};
+    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(path));
+    ASSIGN_OR_RETURN(auto input_file, fs->new_random_access_file(opts, path));
+    ASSIGN_OR_RETURN(auto serialized_string, input_file->read_all());
```

Comment:
If aggregate tablet meta is too large, we should avoid read all at once.

Reply:
The workflow should be:
1. get the length of aggregate tablet meta.
2. if length < N, then use get_all to read whole aggregate tablet meta, and then parse specific tablet meta from it.
3. if length >= N, then just load the footer `SharedTabletMetadataPB`, and find the <offset,size> for specific tablet, then read this tablet meta.

Reply:
> The workflow should be:
> 
> 1. get the length of aggregate tablet meta.
> 2. if length < N, then use get_all to read whole aggregate tablet meta, and then parse specific tablet meta from it.
> 3. if length >= N, then just load the footer `SharedTabletMetadataPB`, and find the <offset,size> for specific tablet, then read this tablet meta.

Yes, but don't not read all at once will increase the api calls of `get_tablet_metadata` and i am not sure which one is better. I think we can first implement it using the `read_all` approach and conduct follow-up testing, then optimize based on the test results.